### PR TITLE
Adding more information to in-memory caching

### PR DESCRIPTION
**Pull request by Rike-cz**
octobercms/october#4715

### CODE REVIEW

File: database-model.md [hunk 1]
```diff
@@ -164,7 +164,7 @@ When attributes names are passed to the `$jsonable` property, the values will be
 
 When requesting data from the database the model will retrieve values primarily using the `get` or `first` methods, depending on whether you wish to [retrieve multiple models](#retrieving-multiple-models) or [retrieve a single model](#retrieving-single-models) respectively. Queries that derive from a Model return an instance of [October\Rain\Database\Builder](../api/october/rain/database/builder).
 
-> **Note**: All model queries have [in-memory caching enabled](../database/query#caching-queries) by default.
+> **Note**: All model queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. Use `reload` method to flush this cache.
```
Comment (w20k) [inline database-model.md]:
```suggestion
> **Note**: All model queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. Use `reload` function to flush in-memory cache.
```

File: database-relations.md [hunk 2]
```diff
@@ -54,6 +54,8 @@ Accessing a relationship as a property is also possible:
 
     $user->posts;
 
+> **Note**: All relationship queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. `load` method won't force cache flushing. For reloading relationship cache use `reloadRelations` method or `reload` whole model.
```
Comment (w20k) [inline database-relations.md]:
```suggestion
> **Note**: All relationship queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. `load` method won't force cache to flush. To reload the relations cache use `reloadRelations` method or `reload` the whole model.
```

**Comment by w20k** [comment]
LGTM, @Rike-cz. 
We have to wait till @bennothommo or @LukeTowers could read-through)

**Comment by Rike-cz** [comment]
@w20k Thanks for helping!

File: database-model.md [hunk 3]
```diff
@@ -164,7 +164,7 @@ When attributes names are passed to the `$jsonable` property, the values will be
 
 When requesting data from the database the model will retrieve values primarily using the `get` or `first` methods, depending on whether you wish to [retrieve multiple models](#retrieving-multiple-models) or [retrieve a single model](#retrieving-single-models) respectively. Queries that derive from a Model return an instance of [October\Rain\Database\Builder](../api/october/rain/database/builder).
 
-> **Note**: All model queries have [in-memory caching enabled](../database/query#caching-queries) by default.
+> **Note**: All model queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. Use `reload` method to flush in-memory cache.
```
Comment (LukeTowers) [inline database-model.md]:
```suggestion
> **Note**: All model queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. While the cache should automatically invalidate itself most of the time, sometimes you will need to use the `$model->reload()` method to flush the cache for more complex use cases.
```

File: database-relations.md [hunk 4]
```diff
@@ -54,6 +54,8 @@ Accessing a relationship as a property is also possible:
 
     $user->posts;
 
+> **Note**: All relationship queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. `load` method won't force cache to flush. To reload the relations cache use `reloadRelations` method or `reload` the whole model.
```
Comment (LukeTowers) [inline database-relations.md]:
```suggestion
> **Note**: All relationship queries have [in-memory caching enabled](../database/query#in-memory-caching) by default. The `load($relation)` method won't force cache to flush. To reload the memory cache use the `reloadRelations()` or the `reload()` methods on the model object.
```

**Pull request merged**